core/app/models/concerns/spree/user_address_book.rb

```diff
@@ -56,16 +56,16 @@ def default_address_attributes=(attributes)
 
       def ship_address=(address)
         be_default = Spree::Config.automatic_default_address
-        save_in_address_book(address.attributes, be_default)
+        save_in_address_book(address.attributes, be_default) if address
       end
 
       def persist_order_address(order)
         if Spree::Config.automatic_default_address
-          save_in_address_book(order.ship_address.attributes, true)
-          save_in_address_book(order.bill_address.attributes, order.ship_address.nil?)
+          save_in_address_book(order.ship_address.attributes, true) if order.ship_address
+          save_in_address_book(order.bill_address.attributes, order.ship_address.nil?) if order.bill_address
```

Comment:
Prefer this. Original change was just re-adding the nil checks that had been previously removed. Thanks